src/components/AddResources/Gemstone.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <v-text-field
+    v-model="formData.color"
+    :counter="10"
+    :rules="nameRules"
+    label="Color"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.carat"
+    :counter="10"
+    :rules="nameRules"
+    label="Carat"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.cut"
+    :counter="10"
+    :rules="nameRules"
+    label="Cut"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.clarity"
+    :counter="10"
+    :rules="nameRules"
+    label="Clarity"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.dimensionX"
+    :counter="10"
+    :rules="nameRules"
+    label="dimensionX"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.dimensionY"
+    :counter="10"
+    :rules="nameRules"
+    label="dimensionY"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.dimensionZ"
+    :counter="10"
+    :rules="nameRules"
+    label="dimensionZ"
+    required
+  ></v-text-field>
+
+  <v-text-field
+    v-model="formData.shape"
+    :counter="10"
+    :rules="nameRules"
+    label="Shape"
+    required
+  ></v-text-field>
+</template>
+
+<script>
+import { useStore } from "vuex";
+
+export default {
+  setup() {
+    const store = useStore();
+    const formData = store.state.resources.resourceDetails;
```

Comment:
I see these are not taken via store getters. Do you know what the difference is and if they should be taken through the getters instead?

Reply:
the only difference is that getters are for more maintainability, organization and if you need to implement some logic.